docs/site/express-with-lb4-rest-tutorial.md

```diff
@@ -0,0 +1,304 @@
+`DsDataSource` as the datasource and `Note` model as the model.
+
+#### Add Note Controller
+
+To complete the `Note` application, create a controller using the
+`lb4 controller note` command, with the `REST Controller with CRUD functions`
+type, `Note` model, and `NoteRepository` repository. The `id`'s type will be
+`number` and base HTTP path name is the default `/notes`.
+
+### Create a Facade Express Application
+
+Create a new file **src/server.ts** to create your Express class:
+
+```sh
+npm install --save express
+npm install --save-dev @types/express
+```
+
+{% include code-caption.html content="src/server.ts" %}
+
+```ts
+import * as express from 'express';
+
+export class ExpressApplication {
+  constructor() {}
+}
+```
+
+Create two properties, the Express application instance and LoopBack application
+instance:
+
+```ts
+import {NoteApplication} from './application';
+import {ApplicationConfig} from '@loopback/core';
+import * as express from 'express';
+
+export class ExpressApplication {
+  private app: express.Application;
+  private lbApp: NoteApplication;
+
+  constructor(options: ApplicationConfig = {}) {
+    this.app = express();
+    this.lbApp = new NoteApplication(options);
+  }
+}
+```
+
+Now, inside the constructor, we're going to add the basepath and expose the
+front-end assets via Express:
+
+```ts
+this.app.use('/api', this.lbApp.requestHandler);
+```
+
+Let's also modify **public/index.html** to update the base path:
+
+{% include code-caption.html content="public/index.html" %}
+
+```html
+<h3>OpenAPI spec: <a href="/api/openapi.json">/openapi.json</a></h3>
+<h3>API Explorer: <a href="/api/explorer">/explorer</a></h3>
+```
+
+Then, we can add some custom Express routes, as follows:
+
+```ts
+import {Request, Response} from 'express';
+import * as path from 'path';
+
+export class ExpressApplication {
+  private app: express.Application;
+  private lbApp: NoteApplication;
+
+  constructor(options: ApplicationConfig = {}) {
+    // earlier code
+
+    // Custom Express routes
+    this.app.get('/', function(_req: Request, res: Response) {
+      res.sendFile(path.resolve('public/express.html'));
+    });
+    this.app.get('/hello', function(_req: Request, res: Response) {
+      res.send('Hello world!');
+    });
+  }
+}
+```
+
+And add the
+[public/express.html](https://github.com/strongloop/loopback-next/blob/master/examples/express-composition/public/express.html)
+file to your project.
+
+Finally, we can add functions to boot the `Note` application and start the
+Express application:
+
+```sh
+npm install --save p-event
+```
+
+```ts
+import * as pEvent from 'p-event';
+
+export class ExpressApplication {
+  private app: express.Application;
+  private lbApp: NoteApplication;
+
+  constructor(options: ApplicationConfig = {}) {
+    //...
+  }
+
+  async boot() {
+    await this.lbApp.boot();
+  }
+
+  async start() {
+    const server = this.app.listen(3000);
```

Comment:
This probably has been asked before, but why don't we need to start the `lb4app` instance as well?

Reply:
Before it was causing some errors with routing (and it worked without it), but looks like it was resolved so I'll add it back.